commit ec189da923fb06acb0ee38e2d0ee55239995e9d6
Author: Justin Stitt <justinstitt@google.com>
Date:   Fri Oct 20 23:18:59 2023 +0000

    rpmsg: Replace deprecated strncpy with strscpy_pad
    
    strncpy() is deprecated for use on NUL-terminated destination strings
    [1] and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    We expect chinfo.name to be NUL-terminated based on its use with format
    strings:
    |       dev_err(&ctrldev->dev, "failed to create %s channel\n", chinfo.name);
    
    Since chinfo is not default initialized, we should NUL-pad the `name`
    field so that the behavior is consistent with what strncpy() provides:
    |       struct rpmsg_channel_info chinfo;
    
    Considering the above, a suitable replacement is `strscpy_pad` due to
    the fact that it guarantees both NUL-termination and NUL-padding on the
    destination buffer.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Link: https://lore.kernel.org/r/20231020-strncpy-drivers-rpmsg-rpmsg_ns-c-v1-1-99b16b00c36c@google.com
    Signed-off-by: Mathieu Poirier <mathieu.poirier@linaro.org>

diff --git a/drivers/rpmsg/rpmsg_ns.c b/drivers/rpmsg/rpmsg_ns.c
index c70ad03ff2e9..bde8c8d433e0 100644
--- a/drivers/rpmsg/rpmsg_ns.c
+++ b/drivers/rpmsg/rpmsg_ns.c
@@ -50,7 +50,7 @@ static int rpmsg_ns_cb(struct rpmsg_device *rpdev, void *data, int len,
 	/* don't trust the remote processor for null terminating the name */
 	msg->name[RPMSG_NAME_SIZE - 1] = '\0';
 
-	strncpy(chinfo.name, msg->name, sizeof(chinfo.name));
+	strscpy_pad(chinfo.name, msg->name, sizeof(chinfo.name));
 	chinfo.src = RPMSG_ADDR_ANY;
 	chinfo.dst = rpmsg32_to_cpu(rpdev, msg->addr);